lib/content-services/src/lib/common/services/saved-searches.service.spec.ts

```diff
@@ -72,136 +71,142 @@ describe('SavedSearchesService', () => {
+            spyOn(localStorage, 'getItem').and.callFake(() => 'true');
+            service.init();
+
+            service.getSavedSearches().subscribe((searches) => {
+                expect(localStorage.getItem).toHaveBeenCalledWith(LOCAL_STORAGE_KEY);
+                expect(service.preferencesApi.getPreference).toHaveBeenCalledWith('-me-', 'saved-searches');
+                expect(searches.length).toBe(2);
+                expect(searches[0].name).toBe('Search 1');
+                expect(searches[1].name).toBe('Search 2');
+                done();
+            });
         });
-    });
 
-    it('should save a new search', (done) => {
-        spyOn(authService, 'getUsername').and.callFake(() => testUserName);
-        spyOn(localStorage, 'getItem').and.callFake(() => 'true');
-        const newSearch = { name: 'Search 3', description: 'Description 3', encodedUrl: 'url3' };
-        service.init();
+        it('should automatically migrate saved searches if config.json file exists', (done) => {
+            spyOn(localStorage, 'setItem');
+            spyOn(authService, 'getUsername').and.callFake(() => testUserName);
 
-        service.saveSearch(newSearch).subscribe(() => {
-            expect(service.preferencesApi.updatePreference).toHaveBeenCalledWith('-me-', 'saved-searches', jasmine.any(String));
-            expect(service.savedSearches$).toBeDefined();
-            done();
+            service.getSavedSearches().subscribe((searches) => {
+                expect(service.nodesApi.getNode).toHaveBeenCalledWith('-my-', { relativePath: 'config.json' });
+                expect(service.nodesApi.getNodeContent).toHaveBeenCalledWith(testNodeId);
+                expect(localStorage.setItem).toHaveBeenCalledWith(LOCAL_STORAGE_KEY, 'true');
+                expect(service.preferencesApi.updatePreference).toHaveBeenCalledWith('-me-', 'saved-searches', SAVED_SEARCHES_CONTENT);
+                expect(service.nodesApi.deleteNode).toHaveBeenCalledWith(testNodeId, { permanent: true });
+                expect(searches.length).toBe(2);
+                done();
+            });
         });
-    });
-
-    it('should emit initial saved searches on subscription', (done) => {
-        spyOn(authService, 'getUsername').and.callFake(() => testUserName);
-        spyOn(localStorage, 'getItem').and.returnValue('true');
-        service.init();
 
-        service.savedSearches$.pipe().subscribe((searches) => {
-            expect(searches.length).toBe(2);
-            expect(searches[0].name).toBe('Search 1');
-            expect(service.preferencesApi.getPreference).toHaveBeenCalledWith('-me-', 'saved-searches');
-            done();
-        });
+        it('should save a new search', (done) => {
+            spyOn(authService, 'getUsername').and.callFake(() => testUserName);
+            spyOn(localStorage, 'getItem').and.callFake(() => 'true');
+            const newSearch = { name: 'Search 3', description: 'Description 3', encodedUrl: 'url3' };
```

Comment:
please set one of those types to this const: 
```ts
SavedSearch
```
or
```ts
Pick<SavedSearch, 'name' | 'description' | 'encodedUrl'>

Reply:
I was thinking rather for a case if in future this test would extend and be more complex (as logic may always change) and then that const may be used in more places but your point is valid too so I will close that comment and approve PR